compiler/tests-common-jvm6/tests/org/jetbrains/kotlin/test/clientserver/TestProcessServer.kt

```diff
@@ -28,22 +28,37 @@ import java.util.concurrent.Executors
 import java.util.concurrent.ScheduledFuture
 import java.util.concurrent.TimeUnit
 
-fun getGeneratedClass(classLoader: ClassLoader, className: String): Class<*> {
+private fun ClassLoader.loadClassOrNull(name: String): Class<*>? =
     try {
-        return classLoader.loadClass(className)
+        loadClass(name)
+    } catch (e: ClassNotFoundException) {
+        null
     }
-    catch (e: ClassNotFoundException) {
-        error("No class file was generated for: " + className)
-    }
-}
 
-fun getBoxMethodOrNull(aClass: Class<*>): Method? {
+private fun Class<*>.getMethodOrNull(name: String, vararg parameterTypes: Class<*>): Method? =
     try {
-        return aClass.getMethod("box")
+        getMethod(name, *parameterTypes)
+    } catch (e: NoSuchMethodException) {
+        null
     }
-    catch (e: NoSuchMethodException) {
-        return null
+
+fun getGeneratedClass(classLoader: ClassLoader, className: String): Class<*> =
+    classLoader.loadClassOrNull(className) ?: error("No class file was generated for: $className")
+
+fun getBoxMethodOrNull(aClass: Class<*>): Method? =
+    aClass.getMethodOrNull("box")
+        ?: aClass.classLoader.loadClassOrNull("kotlin.coroutines.Continuation")?.let { aClass.getMethodOrNull("box", it) }
+        ?: aClass.classLoader.loadClassOrNull("kotlin.coroutines.experimental.Continuation")?.let { aClass.getMethodOrNull("box", it) }
+
+fun runBoxMethod(method: Method): String {
+    if (method.parameterCount == 0) {
+        return method.invoke(null) as String
```

Comment:
Now that I think about it I'm confused that this cast doesn't fail in debugger tests, where `box` returns `Unit`.

Reply:
Looks like there are only 2 types of `box` methods in tests now:
`fun box(): String` and  `fun box()`, second one compiled as `void box() `. And I assume that JVM return null for `void` functions through `invoke` invocation. But yes definitely it could fail on other result types

Reply:
Done, I think...? I moved the decision of nullability of the result type to the consumer of runBoxMethod. But perhaps this should just be generalized to `Any`. Thoughts?

Reply:
Oh...it's Kotlin, I've looked at original Java code, sorry for missleading

Reply:
Indeed, some investigation is required why there was no fail